### PR TITLE
(DOCSP-23993): Realm Flipper

### PR DESCRIPTION
**Pull request by mongomoe**
### Jira

- https://jira.mongodb.org/browse/DOCSP-23993

### Staged Changes (Requires MongoDB Corp SSO)

- [Debugging with Flipper - React Native SDK](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-23993-realm-flipper/sdk/react-native/examples/debugging-with-flipper/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+Debugging with Flipper - React Native SDK
+=========================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+To debug Realm apps built with React Native, use the Flipper Debugger. Flipper provides a way to visualize, inspect, and control your apps from a simple desktop interface. In addition to Flipper's intuitive interface, the Realm Flipper Plugin provides additional to improve your app debugging experience:
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:14]: > provides additional __________ to improve your app debugging experience

missing words here

Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:14]: I know we link to the Flipper Getting Started guide in Step 1, but thoughts on adding links here in the overview to [Flipper](https://fbflipper.com/) and the [Realm Flipper Plugin](https://github.com/realm/realm-flipper-plugin) repo?

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+- Live Objects: See objects in real-time.
+- Schema Dependency Graph: View schemas and their relationships.
+- Make changes: Create, modify and delete objects
+- Query: Query the database using RQL
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:19]: Right now the page really only tells you how to set up Flipper + Realm plugin but not how to use the plugin.

It might be nice to expand these into full sections that tell/show you how to do each of them in Flipper. It's a bit of documenting someone else's UI, but they're features that ship with our plugin.

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      - `Manual Android Setup <https://fbflipper.com/docs/getting-started/react-native-android/>`_
+      - `Manual iOS Setup <https://fbflipper.com/docs/getting-started/react-native-ios/>`_
+      
+      If you did use ``react-native init`` to initialize your app and are on
+      React Native version ``0.62`` or greater, Flipper integration is supported
+      out of the box. Otherwise, if you are on an earlier version of React Native, follow the `Automatic Setup <https://fbflipper.com/docs/getting-started/react-native/>`_ guide. 
+
+   .. step:: Launch the Flipper Desktop App
+      
+      Simply launch your React Native app, and then open the Flipper Desktop application. Your React Native app should be available in the :guilabel:`App Inspect` tab of the Flipper UI.
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:43]:
```suggestion
      Launch your React Native app, and then open the Flipper Desktop application. Your React Native app should be available in the :guilabel:`App Inspect` tab of the Flipper UI.
```

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      .. figure:: /images/reload-flipper.png
+         :alt: Add Realm Flipper Plugin     
+      
+   .. step:: Install Realm Flipper Plugin Device with NPM
+      
+      :npm:`Realm Flipper Plugin Device <package/realm-flipper-plugin-device>` enables React Native applications to interact with Realm Flipper Plugin. Realm Flipper Plugin Device also requires the :npm:`React Native Flipper plugin <package/react-native-flipper>`. To install both, run the following command:
+
+      .. code-block:: shell
+
+         npm install realm-flipper-plugin-device n
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:67]:
```suggestion
         npm install realm-flipper-plugin-device react-native-flipper
```

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      
+      If you are developing on iOS run the following command to auto-link the ``react-native-flipper`` module with CocoaPods:
+
+      .. code-block:: shell
+
+         cd ios
+         pod install
+         cd ..
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:75]: Andrew suggested we use https://www.npmjs.com/package/pod-install to do this in the tutorial, maybe we should here too?

`npx pod-install`

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+   .. step:: Import Realm Flipper Plugin in your Code
+      
+      Import the ``RealmPlugin`` component ``realm-flipper-plugin-device`` package in the file in which you open a realm.
+
+      .. code-block:: javascript
+
+         import RealmPlugin from 'realm-flipper-plugin-device';
+
+      Within the return statement of render your ``RealmPlugin`` and pass it the
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:85]: > Within the return statement of ________, render your ``RealmPlugin``...

Comment (mongomoe) [reply source/sdk/react-native/examples/debugging-with-flipper.txt]: Changed to:  Within your component's return statement, render your ``RealmPlugin`` and pass it the opened realm as a parameter.

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 7]
```diff
@@ -0,0 +1,121 @@
+.. _react-native-client-debugging-with-flipper:
+.. _react-native-query-data:
```
Comment (nlarew) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:2]:
```suggestion
```

This is giving a duplicate target error and doesn't seem relevant to the page

File: source/sdk/react-native/examples/debugging-with-flipper.txt (hunk 5 above)
Comment (nlarew) [reply source/sdk/react-native/examples/debugging-with-flipper.txt:80]: To clarify, you shouldn't `cd` if you use the `npx` command. i.e. it's either
```
cd ios && pod install && cd ..
```

or just

```
npx pod-install
```

^ Also note the package name has a dash in it

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+   .. step:: Import Realm Flipper Plugin in your Code
+      
+      Import the ``RealmPlugin`` component ``realm-flipper-plugin-device`` package in the file in which you open a realm.
+
+      .. code-block:: javascript
+
+         import RealmPlugin from 'realm-flipper-plugin-device';
+
+      Within your component's return statement, render your ``RealmPlugin`` and pass it the
+      opened realm as a parameter.
```
Comment (kneth) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:91]: Maybe it is worth mentioning that you can pass in as many Realms as you want. Many apps are using multiple Realms.

File: source/sdk/react-native/examples/debugging-with-flipper.txt [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+   .. step:: Launch the Flipper Desktop App
+      
+      Launch your React Native mobile app and open the Flipper Desktop application.
+      Your React Native app should be available in Flipper's :guilabel:`App Inspect` tab.
+      
+      .. figure:: /images/flipper-getting-started.png
+         :alt: Flipper Getting Started
+
+   .. step:: Add the Realm Flipper Plugin to the Flipper Desktop App
```
Comment (kneth) [inline source/sdk/react-native/examples/debugging-with-flipper.txt:52]: I don't where it should go in, but in order to be able to debug (i.e. setting break points, single stepping, etc.) you must use Hermes as engine. And that will require that you use Realm JS v11.0.0 or later.

**Review by kneth — COMMENTED** [review]
The Realm Flipper plugin is technically not about debugging but inspecting your database (https://fbflipper.com/docs/features/plugins/databases/ is the proper category). The purist in me isn't so happy when I read "To debug Realm apps" :-)
